### PR TITLE
Update response model

### PR DESCRIPTION
**Pull request by yurypat**

### CODE REVIEW

File: src/Mews.Fiscalization.Greece/Model/Result/SendInvoiceError.cs [hunk 1]
```diff
@@ -1,19 +1,41 @@
...
         public string Message { get; }
+
+        private ErrorCode MapErrorCode(string code)
+        {
+            switch (code)
+            {
+                case SendInvoiceErrorCodes.InternalServerErrorCode:
+                case "TechnicalError":
+                    return ErrorCode.InternalServerError;
+
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalization.Greece/Model/Result/SendInvoiceError.cs:25]: I would remove the empty lines between the cases.

File: src/Mews.Fiscalization.Greece/Model/Result/SendInvoiceError.cs [hunk 2]
```diff
@@ -1,19 +1,41 @@
...
+                case SendInvoiceErrorCodes.TimeoutErrorCode:
+                    return ErrorCode.Network;
+
+                case SendInvoiceErrorCodes.ForbiddenErrorCode:
+                    return ErrorCode.InvalidCredentials;
+
+                case "ValidationError":
+                case "XMLSyntaxError":
+                    return ErrorCode.UnhandledValidation;
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalization.Greece/Model/Result/SendInvoiceError.cs:34]: `ValidationError`?

File: src/Mews.Fiscalization.Greece/Model/Result/SendInvoiceError.cs [hunk 3]
```diff
@@ -1,19 +1,41 @@
...
+        private ErrorCode MapErrorCode(string code)
+        {
+            switch (code)
+            {
+                case SendInvoiceErrorCodes.InternalServerErrorCode:
+                case "TechnicalError":
+                    return ErrorCode.InternalServerError;
+
+                case SendInvoiceErrorCodes.TimeoutErrorCode:
+                    return ErrorCode.Network;
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalization.Greece/Model/Result/SendInvoiceError.cs:27]: `NetworkError`

File: src/Mews.Fiscalization.Greece/Model/Result/SendInvoiceError.cs [hunk 4]
```diff
@@ -1,19 +1,41 @@
...
 
         public string Message { get; }
+
+        private ErrorCode MapErrorCode(string code)
+        {
+            switch (code)
+            {
+                case SendInvoiceErrorCodes.InternalServerErrorCode:
+                case "TechnicalError":
+                    return ErrorCode.InternalServerError;
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalization.Greece/Model/Result/SendInvoiceError.cs:24]: `TechnicalError`

File: src/Mews.Fiscalization.Greece/RestClient.cs [hunk 5]
```diff
@@ -75,5 +98,30 @@ private HttpRequestMessage BuildHttpPostMessage(string messageContent)
 
             return message;
         }
+
+        private ResponseDoc BuildResponseDocWithErrors(string errorCode, string errorMessage, int invoicesCount)
+        {
+            var invoiceResults = new List<Response>();
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalization.Greece/RestClient.cs:104]: Please use linq (select).

File: src/Mews.Fiscalization.Greece/Model/Result/SendInvoiceError.cs (hunk 4 above)
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/Result/SendInvoiceError.cs:24]: Will make all changes to this file, thanks

File: src/Mews.Fiscalization.Greece/RestClient.cs (hunk 5 above)
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/RestClient.cs:104]: Will change, thanks

File: src/Mews.Fiscalization.Greece/Model/Result/Error.cs [hunk 6]
```diff
@@ -0,0 +1,15 @@
+﻿namespace Mews.Fiscalization.Greece.Model.Result
+{
+    public class Error
```
Comment (KaliCZ) [inline src/Mews.Fiscalization.Greece/Model/Result/Error.cs:3]: I'm not quite sure if we need this class to exist at all, it doesn't do anything public.

Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/Result/Error.cs:3]: It was used to distinguish input type from property in the SendInvoiceResult class, as they have different structures but similar names. As we are returning back to the collection of errors I can remove this class

Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/Result/Error.cs]: Removed

File: src/Mews.Fiscalization.Greece/RestClient.cs (hunk 5 above)
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/RestClient.cs]: Done

File: src/Mews.Fiscalization.Greece/RestClient.cs [hunk 7]
```diff
@@ -99,28 +100,23 @@ private HttpRequestMessage BuildHttpPostMessage(string messageContent)
...
+                            }
                         }
-                    }
-                });
-            }
-
-            return new ResponseDoc
-            {
-                Responses = invoiceResults.ToArray()
+                    }).ToArray()
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalization.Greece/RestClient.cs:119]: pls fix the indentation (spaces), it should be aligned with `Responses`

**Review by abdallahbeshi — APPROVED** [review]

**Review by KaliCZ — APPROVED** [review]

File: src/Mews.Fiscalization.Greece/RestClient.cs (hunk 7 above)
Comment (abdallahbeshi) [reply src/Mews.Fiscalization.Greece/RestClient.cs:119]: I fixed it 👍

Comment (yurypat) [reply src/Mews.Fiscalization.Greece/RestClient.cs:119]: Thanks!

**Pull request merged**